### PR TITLE
Docs fixes2

### PR DESCRIPTION
**Pull request by jechter**
A batch of docs updates. Also fixed a minor UI bug in the Input Actions editor window I ran into.

### CODE REVIEW

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md [hunk 1]
```diff
@@ -28,41 +23,45 @@ The debugger shows a tree breakdown of the state of the input system.
...
-|Settings||
-|Metrics||
+|Settings|Shows the currently active Input System [settings](Settings.md).|
+|Metrics|Shows some statics about Input System resource usage.|
 
-### Debugging Remotely
+### Debugging Devices
 
-You can connect input debugger to a player running on a device. This makes input activity from the player observable in the editor. The mechanism used by this is Unity's `PlayerConnection`, i.e. the same mechanism by which the Unity profiler can be connected to a player.
+You can double-click on any [input device](Devices.md) in the `Devices` list in the debugger window to open a window showing information about the device, including real-time state information for its controls.
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Documentation~/Debugging.md:31]: These days there's also the option to right-click a device. Gives some additional options.

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md [hunk 2]
```diff
@@ -28,41 +23,45 @@ The debugger shows a tree breakdown of the state of the input system.
...
 
-    ////TODO: explain "Remote Devices"
+The top of the device window shows some general information about the specific device (such as name, manufacturer, serial number, etc).
 
-### Debugging Devices
+Below that, you you see a view listing the devices controls, and their state. This is very useful when debugging input issues, as it allows you to verify if the data the input system is receiving from the input device is what you'd expect it to be in the first place. At the top of this view you will see the following buttons:
+
+* `State`: Show the current state of the device in a new window. This is identical to the information as shown in this view, but is not updated, so you can take a slice of input state data and take the time to inspect it as needed.
+
+* `HID Descriptor`: Only shown for devices connected using the HID protocol. This will open a window showing the detailed HID specifications for the device and each of it's logical controls.
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Documentation~/Debugging.md:41]: Probably a good place to link to HID.md.

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md [hunk 3]
```diff
@@ -28,41 +23,45 @@ The debugger shows a tree breakdown of the state of the input system.
...
 
->NOTE: For an alternative way to debug/visualize controls and/or devices, see [Visualizing Controls](#visualizing-controls).
+Finally, in the bottom of the window, you can inspect the event stream coming from the device. Any [Input Event](Events.md) generated by the device will be shown here. You can double-click any event in the list to inspect the full device state at the time of the event.
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Documentation~/Debugging.md:43]: There's also the option to select two (IIRC actually any number >=2) events and right-click and select "Compare" to see only differences between the state payloads.

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md [hunk 4]
```diff
@@ -28,41 +23,45 @@ The debugger shows a tree breakdown of the state of the input system.
...
 
 ### Debugging Actions
 
->NOTE: For an alternative way to debug/visualize actions, see [Visualizing Actions](#visualizing-actions).
+Any active [actions](Actions.md) are listed in the Actions list in the debugger window (only shown if any actions are active and the editor is in play mode). If an action has actively bound controls, you can click the arrow next to the action to see them in the list. This is useful to debug whether your bindings correctly map to the controls you want them to bind to. See ["Binding Resolution"](ActionBindings.md#binding-resolution) for more info on how bindings are mapped to controls.
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Documentation~/Debugging.md:47]: Maybe "Any enabled actions" to unambiguously refer to `enabled` and `Enable`.

Comment (Rene-Damm) [reply Packages/com.unity.inputsystem/Documentation~/Debugging.md:47]: Might also be worth pointing out that this view will *not* show enabled actions that belong to InputUsers. Those will be listed with the users instead.

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md [hunk 5]
```diff
@@ -28,41 +23,45 @@ The debugger shows a tree breakdown of the state of the input system.
...
+The [`Layouts`](Layouts.md) list in the Debugger window shows a breakdown of all registered [control and device layouts](Layouts.md). This is essentially the database of supported hardware and the knowledge of how to represent a given piece of input hardware. This is mostly handy when you want to [create a new device mapping](HowDoI.md#-create-my-own-custom-devices), and want to see how it gets represented by the input system.
 
->NOTE: The input visualizer components are only available in the editor and in development players.
+### Debugging Remotely
 
-### Visualizing Controls
+You can connect input debugger to a player running on a device. This makes input activity from the player observable in the editor. The mechanism used by this is Unity's `PlayerConnection`, i.e. the same mechanism by which the Unity profiler can be connected to a player.
 
-### Visualizing Actions
+>NOTE: At the moment, debugging input in players is restricted to seeing devices and events from connected players. There is no support yet for seeing other input-related data from players.
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Documentation~/Debugging.md:63]: Might be worth adding something like "such as actions and input users." for extra explicitness.

File: Packages/com.unity.inputsystem/Documentation~/Settings.md [hunk 6]
```diff
@@ -18,43 +18,45 @@ There are three distinct ways in which the input system processes input.
...
-|Dynamic Update|Events are processed in irregular intervals governed by the current framerate.|
-|Manual Update|Events are not processed automatically by instead are flushed out whenever the user calls `InputSystem.Update()`.|
+|[`Fixed Update`](../api/UnityEngine.InputSystem.InputSettings.UpdateMode.html)|Events are processed in intervals of fixed length. This corresponds to how [`MonoBehaviour.FixedUpdate`](https://docs.unity3d.com/ScriptReference/MonoBehaviour.FixedUpdate.html) operates. The length of each interval is determined by [`Time.fixedDeltaTime`](https://docs.unity3d.com/ScriptReference/Time-fixedDeltaTime.html).|
+|[`Dynamic Update`](../api/UnityEngine.InputSystem.InputSettings.UpdateMode.html)|Events are processed in irregular intervals governed by the current framerate.|
+|[`Manual Update`](../api/UnityEngine.InputSystem.InputSettings.UpdateMode.html)|Events are not processed automatically by instead are flushed out whenever the user calls [`InputSystem.Update()`](../api/UnityEngine.InputSystem.InputSystem.html#UnityEngine_InputSystem_InputSystem_Update).|
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Documentation~/Settings.md:23]: Typo in the original... `by` -> `but`

File: Packages/com.unity.inputsystem/Documentation~/Testing.md [hunk 7]
```diff
@@ -173,4 +173,4 @@ Alternatively, arbitrary input events can be fed into the system and arbitrary i
     }
 ```
 
->NOTE: For reference, the tests for the input system itself can be found [here](https://github.com/Unity-Technologies/InputSystem/tree/stable/Packages/com.unity.inputsystem/Tests/InputSystem).
+>NOTE: For reference, the tests for the input system itself can be found [here](https://github.com/Unity-Technologies/InputSystem/tree/stable/Assets/Tests/InputSystem).
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Documentation~/Testing.md:176]: I think the assembly setup here has also changed. Think both the screenshot and the test still refer to the obsolete old name (`Unity.InputSystem.Tests.dll`?).

File: Packages/com.unity.inputsystem/InputSystem/Editor/Settings/InputSettingsProvider.cs [hunk 8]
```diff
@@ -77,11 +77,6 @@ public override void OnGUI(string searchContext)
 
             using (new EditorGUI.DisabledScope(m_AvailableInputSettingsAssets.Length == 0))
             {
-                EditorGUILayout.HelpBox(
-                    "Please note that the new input system is still under development and not all features are fully functional or stable yet.\n\n"
-                    + "For more information, visit https://github.com/Unity-Technologies/InputSystem or https://forum.unity.com/forums/new-input-system.103/.",
-                    MessageType.Warning);
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Editor/Settings/InputSettingsProvider.cs:83]: Heh, showing some good old faith here :D

**Review by Rene-Damm — APPROVED** [review]

**Comment by jechter** [comment]
Will implement all of Rene's feedback in the next batch. Seems the tests it says it is waiting for are actually green.

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md (hunk 1 above)
Comment (jechter) [reply Packages/com.unity.inputsystem/Documentation~/Debugging.md:31]: I actually disagree with this. While a good UI should implement all commonly expected gestures to interact with an item, I don't think a good documentation should always list  all possible paths to get there (at least not when the other paths don't add useful functionality unique to those paths). IMO that makes the text harder to read, and does not add much useful information.

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md (hunk 2 above)
Comment (jechter) [reply Packages/com.unity.inputsystem/Documentation~/Debugging.md:41]: Will do!

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md (hunk 3 above)
Comment (jechter) [reply Packages/com.unity.inputsystem/Documentation~/Debugging.md:43]: Oh, did not know about that. Will add it.

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md (hunk 4 above)
Comment (jechter) [reply Packages/com.unity.inputsystem/Documentation~/Debugging.md:47]: done.

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md (hunk 5 above)
Comment (jechter) [reply Packages/com.unity.inputsystem/Documentation~/Debugging.md:63]: Done

File: Packages/com.unity.inputsystem/Documentation~/Testing.md (hunk 7 above)
Comment (jechter) [reply Packages/com.unity.inputsystem/Documentation~/Testing.md:176]: Will get to this when updating screenshots

File: Packages/com.unity.inputsystem/Documentation~/Debugging.md (hunk 1 above)
Comment (Rene-Damm) [reply Packages/com.unity.inputsystem/Documentation~/Debugging.md:31]: Sure, no problem with that :)

**Pull request merged**